src/index.d.ts

```diff
@@ -0,0 +1,58 @@
+declare type elementGenerator = (
+	stream: Renderer.Stream,
+	nonBlocking: boolean,
+	callback: () => void
+) => void;
+
+interface StatelessFunctionalComponent<P> {
```

Comment:
`<P>`: I'm not keen on single-char variables, but would grudgingly accept that here (and below) if it's a TypeScript convention.

Reply:
I was inspired by this here: https://www.typescriptlang.org/docs/handbook/generics.html

Reply:
Fair enough - as discussed elsewhere, it should be consistent though, i.e. `<T>`.

Reply:
done